pkg/core/notary_test.go

```diff
@@ -91,6 +101,22 @@ func TestNotary(t *testing.T) {
 		completedTxes[tx.Hash()] = tx
 		return nil
 	}
+	getCompletedTx := func(t *testing.T, waitForNonNil bool, h util.Uint256) *transaction.Transaction {
+		if !waitForNonNil {
+			mtx.RLock()
+			defer mtx.RUnlock()
+			return completedTxes[h]
+		}
+
+		var completedTx *transaction.Transaction
+		require.Eventually(t, func() bool {
+			mtx.RLock()
+			defer mtx.RUnlock()
+			completedTx = completedTxes[h]
+			return completedTx != nil
+		}, time.Second*3, time.Millisecond*50, errors.New("main transaction expected to be completed"))
```

Comment:
I'm a bit worried about this test, it's still failing from time to time:
```
2021-07-21T13:30:09.1623849Z     logger.go:130: 2021-07-21T13:29:54.400Z	ERROR	new transaction callback finished with error	{"error": "error while finalizing transaction"}
2021-07-21T13:30:09.1625612Z     logger.go:130: 2021-07-21T13:29:54.760Z	INFO	persisted to disk	{"blocks": 43, "keys": 554, "headerHeight": 93, "blockHeight": 93, "took": "273.203µs"}
2021-07-21T13:30:09.1626622Z     notary_test.go:112: 
2021-07-21T13:30:09.1627098Z         	Error Trace:	notary_test.go:112
2021-07-21T13:30:09.1627688Z         	            				notary_test.go:341
2021-07-21T13:30:09.1628275Z         	            				notary_test.go:574
2021-07-21T13:30:09.1628767Z         	Error:      	Condition never satisfied
2021-07-21T13:30:09.1629373Z         	Test:       	TestNotary
2021-07-21T13:30:09.1630090Z         	Messages:   	main transaction expected to be completed
2021-07-21T13:30:09.1630937Z --- FAIL: TestNotary (5.66s)
2021-07-21T13:30:09.1645776Z === RUN   TestCreateResponseTx
```
Although code looks good to me and the problem is solved.

Reply:
I wasn't able to reproduce it locally, but increasing wait time seems to help. Probably it is related to built-time quotas on CI.